app/extensions/ipfs/manifest.json

```diff
@@ -0,0 +1,18 @@
+{
+  "manifest_version": 2,
+  "name": "IPFS",
+  "version": "0.1.0",
+  "default_locale": "en",
+  "description": "InterPlanetary FileSystem",
+  "icons": {
+    "128": "img/ipfs-128.png",
+    "48": "img/ipfs-48.png",
+    "16": "img/ipfs-16.png"
+  },
+  "offline_enabled": true,
+  "background": {
+    "page": "ipfs.html",
+    "persistent": true
+  },
+  "content_security_policy": "script-src 'self' 'unsafe-eval'"
```

Comment:
I believe the csp you have defined in the page above can be defined here instead

Reply:
if the CSP is just for the IPFS extension, it should be defined here instead of in index-dev.html. i would still encourage refactoring such that unsafe-eval isn't needed.

Reply:
@bridiver when an extension has an index.html page, the extension loader ignores this CSP and applies the one in index.html. So yes, you are right, the index.html one is the only one necessary.

@diracdeltas agreed that we should not have the need for unsafe-eval. We are working on it :)

Reply:
@diasdavid I think the latest version of js-ipfs-api is ready to be tested without evals :)

Reply:
Here we just use `js-ipfs`. It seems that there still something. Need to test this more:

![image](https://user-images.githubusercontent.com/1211152/30164411-5a229d80-93d4-11e7-90e2-34693ed06f3a.png)

Reply:
that is definitely still a protobuf thing

Reply:
looks like we missed two modules

```
npm ls protocol-buffers
ipfs@0.25.4 /Users/dignifiedquire/opensource/ipfs/js-ipfs
├─┬ libp2p-floodsub@0.11.0
│ └─┬ libp2p-crypto@0.9.4
│   └── protocol-buffers@3.2.1  deduped
└─┬ libp2p-kad-dht@0.5.0
  └── protocol-buffers@3.2.1
```

Reply:
💡 So it is evident. Nice, I was worried that it was npm false deduping. Thanks for catching that!

Reply:
![image](https://user-images.githubusercontent.com/1211152/30181615-bec537b6-940b-11e7-9234-4882d597a7a7.png)

Woot 🚀 No more unsafe-eval!

Reply:
🎉  🎉 🎉